### ISSUE TITLE
Offline Tutorial

### ISSUE THREAD

**Issue by txorav** [phil-opp/blog_os#1164]
hello guys
i need an offline version of this tutorial if there is i would likr to have it and thank you

**Comment by bjorn3** [comment]
There are two options:

You could clone this repo and build it yourself. See https://github.com/phil-opp/blog_os/blob/56ec24cbe155d768cc85752bafd6d7bc1982af88/.github/workflows/blog.yml#L22-L33 for the steps necessary for building it. You can then find the pages in `blog/public`.

Alternatively you can take the official build from CI. For example as of writing this https://github.com/phil-opp/blog_os/actions/runs/3274928653 is the last CI run. In the artifacts section there is the generated_site artifact which is a zip file containing the entire site.

**Comment by mcandre** [comment]
Second this feature request, as a way to grow readership.

I don't imagine the ZIP is an ePUB, right? Just a siterip, requiring a full Web browser.

What if we received the articles as a Git Book, which would provide a server generated offline ebook, in suitable formats for popular ereader applications?

**Comment by phil-opp** [comment]
Yes, the zip is the website in HTML format, so you need a web browser to view it.

Unfortunately, we don't have EPUB support right now (see also #1070). I think it should be possible to use calibre and its [`ebook-convert`](https://manual.calibre-ebook.com/generated/en/ebook-convert.html) CLI tool to create a book from the HTML website, but we probably need to do some layout adjustments before doing the conversion.

Switching to GitBook is not planned because I like to keep the blog format.
